ec2_vpc: resource_tags aren't required if vpc id given

willthames opened the pull request:
##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
ec2_vpc

##### ANSIBLE VERSION
```
ansible 2.3.0 (devel b8dd4dd52b) last updated 2017/01/17 16:30:08 (GMT +1000)
  config file = 
  configured module search path = Default w/o overrides
```

##### SUMMARY
It makes little sense to require resource tags if you have
the VPC ID of a VPC that you wish to delete.

willthames commented:
!needs_revision

willthames commented:
Will see if #20344 gets support - this shouldn't be accepted if ec2_vpc should be deprecated, I'll use ec2_vpc_net instead (which really should take a vpc_id!)

gundalow commented on lib/ansible/modules/cloud/amazon/ec2_vpc.py:
```diff
@@ -730,6 +756,9 @@ def main():
     else:
         module.fail_json(msg="region must be specified")
 
+    if not module.params.get('vpc_id') and not module.params.get('resource_tags'):
```
Any reason you are not using the built in `required_one_of` function for this?

willthames replied:
Because it's not at all documented and I'd forgotten about its existence? (even the ec2 helper functions are better documented than `required_one_of`). I've updated the PR now to use it.

gundalow replied:
It's on my to do list to get this stuff documented.

gundalow reviewed (changes requested):
need_revision

willthames commented:
Fixes #19118

Commit:
ec2_vpc: resource_tags aren't required if vpc id given

It makes little sense to require resource tags if you have
the VPC ID of a VPC that you wish to delete.

Improved `route_table` documentation

willthames commented:
@gundalow can you re-review please.

Commit:
Improve ec2_vpc module PEP8 compliance

s-hertel commented:
@willthames Since #20344 was merged is this still something you're wanting to see merged, or should it be closed?

willthames commented:
That's a good question - it is a bug, even if the module is deprecated. I'm fine either way tbh.

s-hertel commented on lib/ansible/modules/cloud/amazon/_ec2_vpc.py:
```diff
@@ -74,7 +74,7 @@
       - Tags in this list are used in conjunction with CIDR block to uniquely identify a VPC in lieu of vpc_id. Therefore,
         if CIDR/Tag combination does not exist, a new VPC will be created.  VPC tags not on this list will be ignored. Prior to 1.7,
         specifying a resource tag was optional.'
```
Probably should add something to the docs. `Prior to 1.7 and after ...`? Will this be backported?

s-hertel commented on lib/ansible/modules/cloud/amazon/_ec2_vpc.py:
```diff
@@ -713,30 +721,32 @@ def terminate_vpc(module, vpc_conn, vpc_id=None, cidr=None):
 
     module = AnsibleModule(
         argument_spec=argument_spec,
+        required_one_of=[['vpc_id', 'resource_tags']],
```
Would a required_if for state absent be better? It's weird to require one of these for creation if you might just plan to use vpc_id to then delete it.

s-hertel commented on lib/ansible/modules/cloud/amazon/_ec2_vpc.py:
```diff
@@ -713,30 +721,32 @@ def terminate_vpc(module, vpc_conn, vpc_id=None, cidr=None):
         module.fail_json(msg='boto required for this module')
 
-    state = module.params.get('state')
+    module.params.get('state')
```
Line not needed?

gundalow commented:
As agreed in 2017-12-07 Core meeting, we've done bulk pep8 fixes across all modules in `devel`.

This was done in https://github.com/ansible/ansible/pull/33682 and https://github.com/ansible/ansible/pull/33690 which have been merged into `devel`. because of this you will need to rebase your PR. Thanks in advance.

biomassives commented:
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  acmeideal@gmail.com
    Domain biomassiv.es has exceeded the max emails per hour (104/100 (104%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from o8.sgmail.github.com ([167.89.101.199]:53785)
	by chi-server32.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES128-GCM-SHA256:128)
	(Exim 4.89)
	(envelope-from <bounces+848413-b37a-greg=biomassiv.es@sgmail.github.com>)
	id 1eNK1f-002m2Q-MS
	for greg@biomassiv.es; Fri, 08 Dec 2017 08:53:19 -0600
DKIM-Signature: v=1; a=rsa-sha1; c=relaxed/relaxed; d=github.com; 
	h=from:reply-to:to:cc:in-reply-to:references:subject:mime-version:content-type:content-transfer-encoding:list-id:list-archive:list-post:list-unsubscribe; 
	s=s20150108; bh=MHNB46nUSmJhuxlMdE/BPWc2tkc=; b=hUWHAm4I9A1DVPGm
	SttDtKq0IXboShFikjgzrJgbvCs77FHhWI1xTzNUjo6Vl5mn9ir8+dbXR5zySLXl
	pabch1YpmX0SkSAdppgKrsVa4F3u1u4Yt6xzwUtMHRtu4aSGiJJC7xhNS3EivRCY
	XwaeHXbttiSU+YfBRoxyq/9O8uk=
Received: by filter0326p1iad2.sendgrid.net with SMTP id filter0326p1iad2-22893-5A2AA738-32
        2017-12-08 14:52:40.848295227 +0000 UTC
Received: from github-smtp2a-ext-cp1-prd.iad.github.net (github-smtp2a-ext-cp1-prd.iad.github.net [192.30.253.16])
	by ismtpd0014p1iad2.sendgrid.net (SG) with ESMTP id wW1qj9_RS72wCA1TUD-DuQ
	for <greg@biomassiv.es>; Fri, 08 Dec 2017 14:52:40.927 +0000 (UTC)
Date: Fri, 08 Dec 2017 14:52:40 +0000 (UTC)
From: John R Barker <notifications@github.com>
Reply-To: ansible/ansible <reply@reply.github.com>
To: ansible/ansible <ansible@noreply.github.com>
Cc: Subscribed <subscribed@noreply.github.com>
Message-ID: <ansible/ansible/pull/20334/c350282015@github.com>
In-Reply-To: <ansible/ansible/pull/20334@github.com>
References: <ansible/ansible/pull/20334@github.com>
Subject: Re: [ansible/ansible] ec2_vpc: resource_tags aren't required if vpc
 id given (#20334)
Mime-Version: 1.0
Content-Type: multipart/alternative;
 boundary="--==_mimepart_5a2aa738bf0e1_421f3f9270216f2897534c";
 charset=UTF-8
Content-Transfer-Encoding: 7bit
Precedence: list
X-GitHub-Sender: gundalow
X-GitHub-Recipient: biomassives
X-GitHub-Reason: subscribed
List-ID: ansible/ansible <ansible.ansible.github.com>
List-Archive: https://github.com/ansible/ansible
List-Post: <mailto:reply@reply.github.com>
List-Unsubscribe: <mailto:unsub+0042d4e264274bc7e2c64f53281c18860785511a56dbc73092cf000000011642693892a169ce0bfddf4a@reply.github.com>,
 <https://github.com/notifications/unsubscribe/AELU4uuRReZZ3-d0LXy9aLnhyMnotVO_ks5s-U04gaJpZM4LlRth>
X-Auto-Response-Suppress: All
X-GitHub-Recipient-Address: greg@biomassiv.es
X-SG-EID: ZlaDthyue/4HG+EgTb7JoZYs48JNb2iQtCcPV6oIs9XZD67TDYDs9pqfxUC18ID+bgYA6LcpKCO1Z9
 l15Z84KVrf8K0Pt4ZgeWlbZlw7Q0KO9iAcZyPeTgJB2+3SZZKuMBl+CDPJEJK9+RMdIAv0P+YqWfii
 0OM25W2suYExN458U09dBTErbrdkzBF87ndP+2fkfEguoG+WY8QczaENCuSUYcB2AIciRLrdsx9Lje
 Q=

----==_mimepart_5a2aa738bf0e1_421f3f9270216f2897534c
Content-Type: text/plain;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

As agreed in 2017-12-07 Core meeting, we've done bulk pep8 fixes across all modules in `devel`.

This was done in https://github.com/ansible/ansible/pull/33682 and https://github.com/ansible/ansible/pull/33690 which have been merged into `devel`. because of this you will need to rebase your PR. Thanks in advance.

-- 
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub:
https://github.com/ansible/ansible/pull/20334#issuecomment-350282015
----==_mimepart_5a2aa738bf0e1_421f3f9270216f2897534c
Content-Type: text/html;
 charset=UTF-8
Content-Transfer-Encoding: quoted-printable

<p>As agreed in 2017-12-07 Core meeting, we've done bulk pep8 fixes across =
all modules in <code>devel</code>.</p>
<p>This was done in <a href=3D"https://github.com/ansible/ansible/pull/3368=
2" class=3D"issue-link js-issue-link" data-error-text=3D"Failed to load iss=
ue title" data-id=3D"280202668" data-permission-text=3D"Issue title is priv=
ate" data-url=3D"https://github.com/ansible/ansible/issues/33682">#33682</a=
> and <a href=3D"https://github.com/ansible/ansible/pull/33690" class=3D"is=
sue-link js-issue-link" data-error-text=3D"Failed to load issue title" data=
-id=3D"280277734" data-permission-text=3D"Issue title is private" data-url=
=3D"https://github.com/ansible/ansible/issues/33690">#33690</a> which have =
been merged into <code>devel</code>. because of this you will need to rebas=
e your PR. Thanks in advance.</p>

<p style=3D"font-size:small;-webkit-text-size-adjust:none;color:#666;">&mda=
sh;<br />You are receiving this because you are subscribed to this thread.<=
br />Reply to this email directly, <a href=3D"https://github.com/ansible/an=
sible/pull/20334#issuecomment-350282015">view it on GitHub</a>, or <a href=
=3D"https://github.com/notifications/unsubscribe-auth/AELU4tBCYSeMfpY_7NxDl=
Q4fbTEoIcWHks5s-U04gaJpZM4LlRth">mute the thread</a>.<img alt=3D"" height=
=3D"1" src=3D"https://github.com/notifications/beacon/AELU4rIMyNSLKRnYduMh1=
C7vXc0UDgsjks5s-U04gaJpZM4LlRth.gif" width=3D"1" /></p>
<div itemscope itemtype=3D"http://schema.org/EmailMessage">
<div itemprop=3D"action" itemscope itemtype=3D"http://schema.org/ViewAction=
">
  <link itemprop=3D"url" href=3D"https://github.com/ansible/ansible/pull/20=
334#issuecomment-350282015"></link>
  <meta itemprop=3D"name" content=3D"View Pull Request"></meta>
</div>
<meta itemprop=3D"description" content=3D"View this Pull Request on GitHub"=
></meta>
</div>

<script type=3D"application/json" data-scope=3D"inboxmarkup">{"api_version"=
:"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"Gi=
tHub"},"entity":{"external_key":"github/ansible/ansible","title":"ansible/a=
nsible","subtitle":"GitHub repository","main_image_url":"https://cloud.gith=
ubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7b=
b5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/1434=
18/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Ope=
n in GitHub","url":"https://github.com/ansible/ansible"}},"updates":{"snipp=
ets":[{"icon":"PERSON","message":"@gundalow in #20334: As agreed in 2017-12=
-07 Core meeting, we've done bulk pep8 fixes across all modules in `devel`.=
\r\n\r\nThis was done in https://github.com/ansible/ansible/pull/33682 and =
https://github.com/ansible/ansible/pull/33690 which have been merged into `=
devel`. because of this you will need to rebase your PR. Thanks in advance.=
"}],"action":{"name":"View Pull Request","url":"https://github.com/ansible/=
ansible/pull/20334#issuecomment-350282015"}}}</script>=

----==_mimepart_5a2aa738bf0e1_421f3f9270216f2897534c--

gundalow commented:
In devel (and therefore 2.5) this module is docs-only
https://github.com/ansible/ansible/blob/a23c950/lib/ansible/modules/cloud/amazon/_ec2_vpc.py#L163-L166

So I wonder if this should be closed, OR is it a candidate for `stable-2.4`?

willthames commented:
This module is disappearing with the next release, so closing this.
